app/controllers/users_controller.rb

```diff
@@ -219,6 +222,10 @@ def update
 		end
 		current_user.save!
 		if params.key?(:api)
+			plan_recipes = current_user.recipes.where(:type => "PlanRecipe")
+			plan_recipes.each do |recipe|
+				recipe.groceries.delete_all
+			end
 			current_user.recipes.where(:type => "PlanRecipe").delete_all
```

Comment:
good idea calling it plan recipes. makes it easier for next group of ppl to continue